docs/content/usage/agit-support.en-us.md

```diff
@@ -47,3 +47,26 @@ Here's another advanced example for creating a new PR targeting `master` with `t
 ```shell
 git push origin HEAD:refs/for/master -o topic="Topic of my PR" -o title="Title of the PR" -o description="# The PR Description\nThis can be **any** markdown content.\n- [x] Ok"
 ```
+
+## Using `git-repo`
+
+If you want to use [`git-repo`](https://git-repo.info/) for your agit workflow, either in single- or multi-repo mode, you need to take a few steps.
+
+1. Create a file in `$GITEA_CUSTOM/public/` called `ssh_info`
+2. Populate it with `{"type":"agit","version":2}`
+3. Add a rewrite rule for your reverse proxy which rewrites the request path equaling `/ssh_info` to `/assets/ssh_info`
```

Comment:
Don't we already provide a mechanism to put files into root, like `robots.txt`?

Reply:
The only one I could find was `/assets/<file>`, and `repo`/`git-repo` expect it to be at the root of the host 🤷🏼‍♀️

Reply:
The quite old "assets" handler logic is: if local file exists, then use it, otherwise, pass it to other handles.

However, the logic has been changed long time ago, the local file doesn't have higher priority than other handlers.

For example: if you have a file in `{Custom}/public/my-name` and an org `my-name`

* For quite old logic (very long time ago): the local file `{Custom}/public/my-name`  is served
    * Users could serve any custom files without reverse proxy tweaks.
* For current logic (has been changed long time ago): the org `my-name` is severed.

TBH, the quite old logic looks good to me, while I guess some nice features were changed unintentionally by many round iterations.

Reply:
The change was introduced by :  add /assets as root dir of public files #15219 in 1.15

After that PR, users couldn't serve their custom files in public anymore.

Reply:
There is special handling for robots.txt to remain under the root path. Maybe we could include the ssh_info in the binary directly? It would require reserving the "ssh_info" username, but I think that would be acceptable as a "breaking" change.

Reply:
I would prefer to take the "quite old logic" back, then end users could still have the ability to serve any file in the web root.

Reply:
Hmm, gitea has defined `/ssh_info` router, but the type has been defined as `gitea` (example: https://try.gitea.io/ssh_info), That's because I'm not sure `agit` flow has been fully implement. looks need more work. and I hasn't checking whether gitea can be used with `git-repo` tool also.

Reply:
> whether gitea can be used with `git-repo` tool also.

I'm working through why it hasn't been working as intended, and this is what I've found so far. If the router has implemented `/ssh_info` but it's wrong, then we should fix it. It's ultimately something from Gerrit:

https://github.com/GerritCodeReview/gerrit/blob/587e12178e06e63c0ac8c31e6ec0cbc59b232eea/java/com/google/gerrit/httpd/raw/SshInfoServlet.java#L31-L48